interface/src/app/formatMissing.filter.js

```diff
@@ -0,0 +1,14 @@
+angular.module('adage.formatMissing.filter', [])
+
+.filter('formatMissing', function() {
+  return function(inputStr, replacementStr) {
+    // default to 'N/A'
+    var replacement = (replacementStr === undefined ? 'N/A' : replacementStr);
+    if (inputStr.replace(/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, '') === '') {
```

Comment:
Can you add some comments to explain what this line is doing? (I guess it is replacing some unprintable UTF-16 characters?)

Reply:
Heh. Sorry about that. It is catching not just plain ASCII whitespace but a couple of other cases. I'll see if I can find the code where I got this example and credit the source.

- [ ] add a comment for this gnarly regex

Reply:
This is actually from the Mozilla documentation for `String.prototype.trim`, found at:

> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim#Polyfill

The compatibility table at the bottom of that page indicates that it "should" be safe to just use `.trim()` in all modern browsers, but I'm being conservative since `.trim()` first appeared in the standard for ES 5.1 and `.replace()` is compatible back to the 3rd edition standard.

I've added a comment to explain the regex. If you think I'm being too conservative, let me know and I'll just use `.trim()`.

Reply:
Thanks. I prefer to be conservative too.